app/main.py

```diff
@@ -1,34 +1,48 @@
+from typing import List
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False
+                 ) -> None:
+        self.is_drowned = is_drowned
+        self.decks = []
+        for row in range(start[0], end[0] + 1):
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        return next(deck for deck in self.decks
+                    if deck.row == row and deck.column == column)
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> None:
+        deck = self.get_deck(row, column)
+        deck.is_alive = False
+        self.is_drowned = all(deck.is_alive is False for deck in self.decks)
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: List[tuple]) -> None:
+        self.field = {}
+        for i in ships:
+            start, end = i
+            ship = Ship(start, end)
+            for le in ship.decks:
+                self.field.update({(le.row, le.column): ship})
+
+    def fire(self, location: tuple) -> str:
+        if location in self.field.keys():
```

Comment:
```suggestion
        if location in self.field:
```
It's getting keys by default